huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go

```diff
@@ -0,0 +1,177 @@
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceDmsKafkaUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	client, err := c.HcDmsV2Client(c.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DMS client: %s", err)
+	}
+
+	instanceUser := d.Get("name").(string)
+	instanceId := d.Get("instance_id").(string)
+
+	createOpts := &model.CreateInstanceUserRequest{
+		InstanceId: instanceId,
+		Body: &model.CreateInstanceUserReq{
+			UserName:   utils.String(instanceUser),
+			UserPasswd: utils.String(d.Get("password").(string)),
+		},
+	}
+
+	_, err = client.CreateInstanceUser(createOpts)
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DMS instance user: %s", err)
+	}
+
+	id := instanceId + "/" + instanceUser
+	d.SetId(id)
+	return resourceDmsKafkaUserRead(ctx, d, meta)
+}
+
+func resourceDmsKafkaUserRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	client, err := c.HcDmsV2Client(c.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DMS client: %s", err)
+	}
+
+	// Split instance_id and user from resource id
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return fmtp.DiagErrorf("invalid id format, must be <instance_id>/<user>")
+	}
+	instanceId := parts[0]
+	instanceUser := parts[1]
+
+	// List all instance users
+	request := &model.ShowInstanceUsersRequest{
+		InstanceId: instanceId,
+	}
+
+	response, err := client.ShowInstanceUsers(request)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error listing DMS instance users")
+	}
+
+	if response.Users != nil && len(*response.Users) != 0 {
+		users := *response.Users
+		for _, user := range users {
+			if *user.UserName == instanceUser {
+				d.Set("instance_id", instanceId)
+				d.Set("name", instanceUser)
+				return nil
+			}
+		}
+	}
+
+	// DB user deleted
+	d.SetId("")
+	log.Printf("[WARN] failed to fetch DMS instance user %s: deleted", instanceUser)
+
+	return nil
+}
+
+func resourceDmsKafkaUserUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	client, err := c.HcDmsV2Client(c.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DMS client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+
+	updateOpts := &model.ResetUserPasswrodRequest{
+		InstanceId: instanceId,
+		UserName:   d.Get("name").(string),
+		Body: &model.ResetUserPasswrodReq{
+			NewPassword: utils.String(d.Get("password").(string)),
+		},
+	}
+
+	log.Printf("[DEBUG] Update DMS instance user options: %#v", updateOpts)
```

Comment:
the plaintext password is printed, the password should be replaced with *** in the log

Reply:
this log removed